src/script/microbit-interfacing/MicrobitConnection.ts

```diff
@@ -4,37 +4,17 @@
  * SPDX-License-Identifier: MIT
  */
 
-import InputBehaviour from '../connection-behaviours/InputBehaviour';
-import MBSpecs from './MBSpecs';
+import { DeviceRequestStates } from '../stores/connectDialogStore';
 
-export interface MicrobitConnection {
-  isSameDevice(other: MicrobitConnection): boolean;
-
-  isConnected(): boolean;
-
-  disconnect(): void;
-
-  listenToInputServices(
-    inputBehaviour: InputBehaviour,
-    inputUartHandler: (data: string) => void,
-  ): Promise<void>;
-
-  listenToAccelerometer(
-    onAccelerometerChanged: (x: number, y: number, z: number) => void,
-  ): Promise<void>;
-
-  listenToButton(
-    buttonToListenFor: MBSpecs.Button,
-    onButtonChanged: (state: MBSpecs.ButtonState, button: MBSpecs.Button) => void,
-  ): Promise<void>;
-
-  listenToUART(onDataReceived: (data: string) => void): Promise<void>;
-
-  setLEDMatrix(matrix: number[][]): Promise<void>;
-
-  setLEDMatrix(matrix: boolean[][]): Promise<void>;
+/**
+ * A connection to a micro:bit.
+ */
+interface MicrobitConnection {
+  connect(...states: DeviceRequestStates[]): Promise<void>;
 
-  setLEDMatrix(matrix: unknown[][]): Promise<void>;
+  reconnect(userTriggered: boolean): Promise<void>;
 
-  getVersion(): MBSpecs.MBVersion;
+  disconnect(userTriggered: boolean): Promise<void>;
```

Comment:
I think we only care about userTriggered inside the bluetooth implementation, not as MicrobitConnection API. Can we remove from the interface and use private versions with the parameter?